File: gherkin/php/src/ParserException/CompositeParserException.php
<?php

declare(strict_types=1);

namespace Cucumber\Gherkin\ParserException;

use Cucumber\Gherkin\ParserException;
use Exception;

final class CompositeParserException extends ParserException
{
    /**
     * @param non-empty-list<ParserException> $errors
     */
    public function __construct(
        public readonly array $errors,
    ) {
        $message = "Parser errors:\n" . join("\n", array_map(fn ($e) => $e->getMessage(), $errors));

        parent::__construct($message, null);
    }
}

File: gherkin/dart/lib/src/exceptions/CompositeParserException.dart
import 'ParserException.dart';

class CompositeParserException extends ParserException
{
  final List<ParserException> errors;

  CompositeParserException(this.errors)
    : super(_getMessage(errors));

  static String _getMessage(List<ParserException> errors) {
    var linedErrors = errors.map((e) => e.message).join('\n');
    return 'Parser errors:\n$linedErrors';
  }

}
